### PR TITLE
New stuff for the chainweb integration

### PR DESCRIPTION
**Pull request by ak3n**
PR checklist:

* [ ] Test coverage for the proposed changes
* [ ] PR description contains example output from repl interaction or a snippet from unit test output
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/kadena-io/pact-core/blob/master/CHANGELOG.md)

Additionally, please justify why you should or should not do the following:

* [ ] Confirm replay/back compat
* [ ] Benchmark regressions
* [ ] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact

### CODE REVIEW

**Comment by ak3n** [comment]
@jmcardon i think it would nice to merge this PR

File: pact/Pact/Core/SPV.hs [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
 import Pact.Core.PactValue
 
 newtype ContProof = ContProof { _contProof :: ByteString }
   deriving (Eq, Ord, Show, Generic)
 
+instance Wrapped ContProof
 
 instance NFData ContProof
 
+instance J.Encode ContProof where
```
Comment (rsoeldner) [inline pact/Pact/Core/SPV.hs:28]: Why do we need an `Encode` instance for `ContProof` ?

File: pact/Pact/Core/SPV.hs [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
 import Control.DeepSeq
+import Control.Lens (Wrapped)
 
 import Pact.Core.DefPacts.Types
 import Pact.Core.PactValue
 
 newtype ContProof = ContProof { _contProof :: ByteString }
   deriving (Eq, Ord, Show, Generic)
 
+instance Wrapped ContProof
```
Comment (rsoeldner) [inline pact/Pact/Core/SPV.hs:24]: What is the reason for it?

File: pact/Pact/Core/Errors.hs [hunk 3]
```diff
@@ -345,6 +345,7 @@ data EvalError
   | EnforcePactVersionParseFailure Text
   | RuntimeRecursionDetected QualifiedName
   | SPVVerificationFailure Text
+  | ContinuationError Text
```
Comment (rsoeldner) [inline pact/Pact/Core/Errors.hs:348]: Should probably be renamed to `SPVContinuationError`?

File: pact/Pact/Core/SPV.hs (hunk 1 above)
Comment (jmcardon) [reply pact/Pact/Core/SPV.hs:28]: Most likely to encode it in outputs

**Review by jmcardon — APPROVED** [review]

**Review by rsoeldner — APPROVED** [review]

File: pact/Pact/Core/Errors.hs (hunk 3 above)
Comment (emilypi) [reply pact/Pact/Core/Errors.hs:348]: Yeah. The defpact cont errors are above, and this is presumably a proof function error from `yield/resume`.

File: pact/Pact/Core/SPV.hs (hunk 2 above)
Comment (emilypi) [reply pact/Pact/Core/SPV.hs:24]: No need.

File: pact/Pact/Core/SPV.hs (hunk 1 above)
Comment (emilypi) [reply pact/Pact/Core/SPV.hs:28]: It wouldn't go into outputs, but it would tx data that we serialize.

**Pull request merged**